obsplus/structures/grid.py

```diff
@@ -0,0 +1,1323 @@
+        # Slice the grid in the Y-direction at the provided X-coordinate
+        dummy = points["Y"][0]
+        ind = grid.get_index((coord, dummy))[0]
+        values = grid.values[ind, :]
+        return points[direction], values
+    else:
+        raise ValueError(f"Unknown direction: {direction}")
+
+
+# ------------------ Coordinate conversion utilities ----------------- #
+def convert_coords(
+    points,
+    conversion,
+    xcol="X",
+    ycol="Y",
+    zcol="Z",
+    xout="X_GRID",
+    yout="Y_GRID",
+    zout="Z_GRID",
+    conversion_kwargs=None,
+    inplace=False,
+):
+    """
+    Converts coordinates from one system to another
+
+    Parameters
+    ----------
+    points : DataFrame (required)
+        DataFrame containing the coordinates of the points to be
+        converted. The following columns are required: ["X", "Y",
+        "Z"(optional)]
+    conversion : list-like or callable (required)
+        Two-dimensional list-like option that specifies the various steps
+        in the conversion process. Values should be of the form
+        ["keyword", value]. Acceptable keywords are: "scale",
+        "translate_x", "translate_y", "translate_z", "rotate_xy",
+        "rotate_xz", "rotate_yz", and "project". Alternatively, a callable
+        that accepts and returns a dataframe can be specified with optional
+        kwargs to handle the conversion.
+    conversion_kwargs : dict (default=None)
+        Used if conversion is a callable. kwargs to be passed to the callable.
+    inplace : bool (default=False)
+        Indicates whether the coordinate conversion should be done on the
+        DataFrame in place or o new DataFrame should be returned.
+
+    Notes
+    -----
+    The following describes the values to be provided with the keywords:\n
+    scale_x : float
+        Scale factor to apply to the x-coordinates\n
+    scale_y : float
+        Scale factor to apply to the x-coordinates\n
+    scale_z : float
+        Scale factor to apply to the x-coordinates\n
+    translate_x : float
+        Amount to move the coordinates in the x-direction\n
+    translate_y : float
+        Amount to move the coordinates in the y-direction\n
+    translate_z : float
+        Amount to move the coordinates in the z-direction\n
+    rotate_xy : float
+        Angle (in radians) to rotate the coordinates in the xy plane.
+        Positive is CCW.\n
+    rotate_xz : float
+        Angle (in radians) to rotate the coordinates in the xz plane.
+        Positive is CCW.\n
+    rotate_yz : float
+        Angle (in radians) to rotate the coordinates in the yz plane.
+        Positive is CCW.\n
+    project : dict
+        Strings for the pyproj.Proj class projection (ex:
+        "+init=EPSG:32611"). Keys in the dictionary should be "from" and
+        "to"\n
+
+    Returns
+    -------
+    pandas Dataframe :
+        Dataframe (either a copy or the original) with the converted coordinates
+    """
+    valid_keys = [
```

Comment:
Maybe we could put this set or tuple at the start of the module? Or the part about conversions at least.